src/pojo-interfaces.test.ts

```diff
@@ -0,0 +1,351 @@
+  const fcBlankNode = fc.constant(null).map(() => DataFactory.blankNode());
+  const fcDefaultGraph = fc.constant(DataFactory.defaultGraph());
+  const fcGraph = fc.oneof(fcDefaultGraph, fcNamedNode);
+  const fcQuadSubject = fc.oneof(fcNamedNode, fcBlankNode);
+  const fcQuadPredicate = fcNamedNode;
+  const fcQuadObject = fc.oneof(fcNamedNode, fcLiteral, fcBlankNode);
+  const fcQuad = fc
+    .tuple(fcQuadSubject, fcQuadPredicate, fcQuadObject, fcGraph)
+    .map(([subject, predicate, object, graph]) =>
+      DataFactory.quad(subject, predicate, object, graph)
+    );
+  const fcTerm = fc.oneof(fcNamedNode, fcLiteral, fcBlankNode, fcQuad);
+  const fcDataset = fc.set(fcQuad).map((quads) => dataset(quads));
+
+  it("can represent all Quads", () => {
+    const blankNode1 = DataFactory.blankNode();
+    const blankNode2 = DataFactory.blankNode();
+    const subject1IriString = "https://some.pod/resource#subject1";
+    const subject1 = DataFactory.namedNode(subject1IriString);
+    const subject2IriString = "https://some.pod/resource#subject2";
+    const subject2 = DataFactory.namedNode(subject2IriString);
+    const predicate1IriString = "https://some.vocab/predicate1";
+    const predicate1 = DataFactory.namedNode(predicate1IriString);
+    const predicate2IriString = "https://some.vocab/predicate2";
+    const predicate2 = DataFactory.namedNode(predicate2IriString);
+    const literalStringValue = "Some string";
+    const literalString = DataFactory.literal(
+      literalStringValue,
+      DataFactory.namedNode(xmlSchemaTypes.string)
+    );
+    const literalLangStringValue = "Some lang string";
+    const literalLangStringLocale = "en-gb";
+    const literalLangString = DataFactory.literal(
+      literalLangStringValue,
+      literalLangStringLocale
+    );
+    const literalIntegerValue = "42";
+    const literalInteger = DataFactory.literal(
+      literalIntegerValue,
+      DataFactory.namedNode(xmlSchemaTypes.integer)
+    );
+    const defaultGraph = DataFactory.defaultGraph();
+    const acrGraphIriString = "https://some.pod/resource?ext=acr";
+    const acrGraph = DataFactory.namedNode(acrGraphIriString);
+
+    const quads = [
+      DataFactory.quad(subject1, predicate1, literalString, defaultGraph),
+      DataFactory.quad(subject1, predicate1, literalLangString, defaultGraph),
+      DataFactory.quad(subject1, predicate1, literalInteger, defaultGraph),
+      DataFactory.quad(subject1, predicate2, subject2, defaultGraph),
+      DataFactory.quad(subject2, predicate1, blankNode1, acrGraph),
+      DataFactory.quad(subject2, predicate1, blankNode2, acrGraph),
+      DataFactory.quad(blankNode1, predicate1, literalString, acrGraph),
+      DataFactory.quad(blankNode2, predicate1, literalString, acrGraph),
+      DataFactory.quad(blankNode2, predicate1, literalInteger, acrGraph),
+      DataFactory.quad(blankNode2, predicate2, literalInteger, acrGraph),
+    ];
```

Comment:
Seems to be a bug here - if I add the single line below here, the test still passes (i.e., as the code is filtering out quads with BNodes as Subject, and only adding them back in if that Subject BNode appears as the object of another quad, but shouldn't this quad appear in the output too...?):
```js
      DataFactory.quad(DataFactory.blankNode(), predicate2, literalInteger, acrGraph),
```

Reply:
Yeah, I addressed that here: https://github.com/inrupt/solid-client-js/pull/947#discussion_r603932077

The tl;dr is: my judgment was that that does not appear in practice (i.e. when would a Dataset include the above Quad?), so I went with Turtle's approach, but full 1:1 correspondence can be achieved by also adding the ability to add labelled blank nodes for "dangling" blank nodes, i.e. expanding

```
    blankNodes: readonly Predicates[];
```

into

```
    blankNodes: readonly Predicates[] | DanglingBlankNode; // With DanglingBlankNode = `_${string}`;
```

Reply:
I realised that we didn't even have to do anything special for Blank Nodes that only appear in Object position, since an empty `Predicates` (i.e. analogous to Turtle's empty `[]`) suffices there, so I only had to add support for them in Subject position: d2cf7c6e3ea74235fea36c1dd3dcce59d0917f9a

If you pull down the latest version and add your example Quad, you should see that it now does appear in the output.

Reply:
As per your request, I also added a test that demonstrates that Lists can be represented: 8bf22219